infrastructure/src/main/java/org/corfudb/infrastructure/FailureHandlerDispatcher.java

```diff
@@ -33,7 +34,8 @@ public void dispatchHandler(IFailureHandlerPolicy failureHandlerPolicy, Layout c
             // Generates a new layout by removing the failed nodes from the existing layout
             Layout newLayout = failureHandlerPolicy.generateLayout(currentLayout, corfuRuntime, failedServers);
             // Seals and increments the epoch.
-            sealEpoch(newLayout);
+            currentLayout.setRuntime(corfuRuntime);
+            sealEpoch(currentLayout);
```

Comment:
Seal should ideally seal the servers which are moving from the old epoch to the new one.
So that there are no messages from the previous epoch trying to make a change after the seal.